src/util/enriched_string.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright (C) 2013 xyz, Ilya Zhuravlev <whatever@xyz.is>
+Copyright (C) 2016 Nore, Nathanaël Courant <nore@mesecons.net>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "enriched_string.h"
+#include "util/string.h"
+#include "log.h"
+EnrichedString::EnrichedString()
+{
+	clear();
+}
+
+EnrichedString::EnrichedString(const std::wstring &string, const std::vector<SColor> &colors):
+	m_string(string),
+	m_colors(colors),
+	m_has_background(false)
+{}
+
+EnrichedString::EnrichedString(const std::wstring &s, const SColor color) {
```

Comment:
Half of these functions have { starting on a new line (like they're supposed to be) and the other half have { on the same line as the function prototype.